lib/Sema/DerivedConformanceRawRepresentable.cpp

```diff
@@ -344,8 +344,9 @@ static bool canSynthesizeRawRepresentable(TypeChecker &tc, Decl *parentDecl,
   auto parentDC = cast<DeclContext>(parentDecl);
   rawType       = parentDC->mapTypeIntoContext(rawType);
 
-  if (!enumDecl->getInherited().empty() &&
-      enumDecl->getInherited().front().isError())
+  auto inherited = enumDecl->getInherited();
+  if (!inherited.empty() && inherited.front().wasValidated() &&
+      inherited.front().isError())
```

Comment:
That's correct. Though, the `RawType` here was an error type anyway — there's no way to recover from that...
This code already exists and hasn't been an issue; I just want this to stop asserting incorrectly. If we'd prefer to take it out, I don't mind.